pkg/cluster/cluster.go

```diff
@@ -37,6 +37,11 @@ import (
 
 // Cluster provides various methods to interact with a cluster.
 type Cluster interface {
+	// Name returns the name of the cluster. It identifies the cluster in the
+	// manager if that is attached to a cluster provider. The value is usually
+	// empty for the default cluster of a manager.
+	Name() string
```

Comment:
We're internally working on a mechanism for dealing with multiple clusters, too. It's been helpful for the cluster identifier to be more flexible than a simple string. We've made it comparable. 

Something like this:
```go
type Cluster[ID comparable] interface {
  Identifier() ID
  ...
}
```

I understand this suggestion would cause a much larger change requiring type parameters in many places. Future work could be done to migrate.

Mostly, I wanted to raise this as something to consider and that calling the method `Identifier` would provide room to migrate in the future.

Reply:
I'm definitely open to that, I think renaming it to `Identifier()` should be easy enough! Maybe some maintainers can leave a note if they consider this a future endeavour and this would be a worthwhile change to prepare that.

Reply:
We were there before, had a logical cluster abstraction, and then went back to string as it just simplifies everything.

@sprsquish Am curious why a string as identifier and some lookup table on the cluster provider side does not work?

Reply:
Also, adding `Name() string` to the interface has no impact on the consumers of the `Cluster` type. But adding a type parameter forces them to pass e.g. string or whatever everywhere including the manager. I don't think that's a good idea. IMO, we should not do it.

Reply:
@sttts 
> Am curious why a string as identifier and some lookup table on the cluster provider side does not work?

Strings work fine.. We like to work with more structure in our identifiers. For instance, working with multiple clusters across cloud providers it's nice to have a structure like this:

```go
type ClusterID struct {
  Provider string
  Region string
  Account string
  Name string
}
```

It can be marshaled into a string, of course, it's just nice not to have to.

I understand the need for a simpler approach and appreciate that this has been discussed already. I think there's a way to migrate over time that follows what's been happening with handlers by introducing `TypedCluster[ID comparable]` and an alias for the common use case `type Cluster = TypedCluster[string]` (Manager would need the same treatment). 

The only change to the current proposal would be to use the more generic `Identifier()` rather than `Name()`. That leaves the door open for future discussions regarding generic identifiers without having to do all the work up front.

Reply:
If I understand your use-case correctly, this pattern should help you elegantly: https://goplay.space/#wRWyxv-GUrj

Or in other words: the type only matters in code that is `ClusterID` aware. Turning the string back into the rich format is one type conversion away.